ugs-platform/ugs-platform-plugin-joystick/src/main/java/com/willwinder/ugs/nbp/joystick/service/JoystickServiceImpl.java

```diff
@@ -133,14 +209,24 @@ private void readData() {
     }
 
     private void notifyListeners() {
-        listeners.forEach(listener -> listener.onUpdate(joystickState));
+        listeners.forEach(listener -> {
+            if (listener == joystickActionDispatcher && !isActionDispatcherActive) {
+                return;
+            }
+
+            listener.onUpdate(joystickState);
+        });
     }
 
     private void updateJoystickAxisState(ControllerAxis controllerAxis) {
         try {
-            // Round values, got issues with the controller having 0.01 as zero-value
-            float value = Math.round(currentController.getAxisState(controllerAxis) * 10) / 10f;
-            JoystickAxis axis = Utils.getJoystickAxisFromControllerAxis(controllerAxis);
+            // We might have rounding errors from the controller, ignore the low value range
```

Comment:
Have you noticed that the analog inputs can change even if the joystick isn't moving? Would it help to keep a moving average?

Reply:
That would probably be part of the `currentController` for analog fields.

Reply:
I have the threshold at 3 percent which was required for my controller. Another user reported that this wasn't enough for their controller. This should probably be configurable as well or if we could come up with a clever algorithm for this as you suggested.